docs/architecture/adr-062-collections-state-layer.md

```diff
@@ -0,0 +1,117 @@
+# ADR 062: Collections, a simplified storage layer for cosmos-sdk modules.
+
+## Changelog
+
+* 30/11/2022: PROPOSED
```

Comment:
```suggestion
* 30/11/2022: Initial draft (@testinginprod)
```

Reply:
Oops... missed this before hitting merge 🤦‍♂️